engine/src/main/java/org/pentaho/di/trans/TransMeta.java

```diff
@@ -1381,6 +1381,10 @@ public StepMeta findPrevStep( StepMeta stepMeta, int nr, boolean info ) {
    * @return The list of the preceding steps
    */
   public List<StepMeta> findPreviousSteps( StepMeta stepMeta, boolean info ) {
+    if ( stepMeta == null ) {
```

Comment:
Could you please add a quick test in TransMetaTest ? I'd rather not see this regress in the future. Thanks.

Reply:
PR updated, test added